src/OrchardCore.Modules/OrchardCore.Forms/Views/Items/ValidationRulePart.cshtml

```diff
@@ -0,0 +1,58 @@
+@using OrchardCore.Forms.Models
+@model ShapeViewModel<ValidationRulePart>
+@{
+    var type = Model.Value.Type;
+    var option = Html.Raw(Model.Value.Option);
+    var formElementPart = Model.Value.ContentItem.As<FormElementPart>();
+    var elementId = formElementPart.Id;
+}
+
+<script asp-src="~/OrchardCore.Forms/Scripts/validator.min.js" debug-src="~/OrchardCore.Forms/Scripts/validator.js" asp-name="validator" depends-on="jQuery" at="Foot"></script>
+
+<script at="Foot">
+    function validateElementFunction(type, originElementId, option) {
+        var elementId = '#' + originElementId;
+        var timestamp = new Date().getTime();
+        var validationElementId = originElementId + timestamp;
+        if (window.validatorObject) {
+            var obj = { type: type, elementId: originElementId, option: option};
+            window.validatorObject.push(obj);
+        }
+        else {
+            window.validatorObject = new Array();
+            var obj = { type: type, elementId: originElementId, option: option};
+            window.validatorObject.push(obj);
+        }
+        $(elementId).bind('change keyup', function (e) {
+            var validateResult = false;
+            if (window.validator.isEmpty(option)) {
+                validateResult = window.validator[type]($(elementId).val());
+            }
+            else if (window.validator.isJSON(option)) {
+                validateResult = window.validator[type]($(elementId).val(), JSON.parse(option));
+            }
+            else {
+                validateResult = window.validator[type]($(elementId).val(), option);
+            }
+            var validationElement = document.getElementById(validationElementId);
+            if (validateResult) {
+                if (validationElement) {
+                    $(elementId).closest('form').find(':submit').removeAttr('disabled');
+                    validationElement.remove();
+                }
+            }
+            else {
+                if (!validationElement) {
+                    var validationMessage =  '@T["{0} validation failed", type]';
```

Comment:
Just a nitpick: Since this looks like this:

![image](https://user-images.githubusercontent.com/1976647/114028761-1ecabd00-9879-11eb-9d17-a2981dc2981f.png)

I'd update it to make it more like a proper sentence:
```suggestion
                    var validationMessage =  '@T["Validation failed for \"{0}\".", type]';
```